Neos.Flow/Classes/ObjectManagement/ObjectManagerInterface.php

```diff
@@ -43,9 +43,10 @@ public function getContext();
      * new keyword and Singleton objects should rather be injected by some type of
      * Dependency Injection.
      *
-     * @param string $objectName The name of the object to return an instance of
-     * @param mixed[] ...$constructorArguments Any number of arguments that should be passed to the constructor of the object
-     * @return object The object instance
+     * @template T of object
+     * @param class-string<T> $objectName The name of the object to return an instance of
```

Comment:
How will this behave with virtual objects that only exist in objects.yaml. May be a non-issue as those are unlikely to show up in any php code but should be considered.

I may be wrong about how virtual objects are handled.

Reply:
Very good point. I wanted to ask the same thing but forgot about it.

It triggers a warning when used with virtual object names it seems: https://phpstan.org/r/62e1adbd-b442-423c-a4d4-4838e9a004b3

Reply:
adjusted ;) cool that this is possible :D